cmd/containerd-shim-runhcs-v1/exec_hcs.go

```diff
@@ -350,7 +339,7 @@ func (he *hcsExec) ForceExit(ctx context.Context, status int) {
 			he.exitFromCreatedL(ctx, status)
 		case shimExecStateRunning:
 			// Kill the process to unblock `he.waitForExit`
-			he.p.Process.Kill(ctx)
+			he.p.Process.Kill(ctx) //nolint:errcheck
```

Comment:
Well these are gonna be a pain :). Need to make it a habit to run the linter before pushing anything

Reply:
If `errcheck` is so annoying, it could possibly be disabled in linter config?

Reply:
I think it's good to keep on to catch some actual missed error checks, I'm mostly just whining. I see there is an option to ignore certain errchecks (`os.Close` and friends that often don't get checked) with `--exclude-use-default`

Reply:
Hmm, it says this is on by default however

Reply:
Well, this PR has 121 suppression of `nolint:errcheck`. I'd say that is... Kinda a lot. So high suppresion rate might lead to "just always suppress it without thinking" behavior.

Reply:
one thing to consider is (of course "within reason", and it may looks a bit ugly in some cases) to explicitly ignore errors for such cases, i.e.;

```suggestion
			_, _ = he.p.Process.Kill(ctx)
```

The `//nolint` comments are not considered by all linters, and (e.g.) some online tools, and some IDEs disregard them (so some linting errors that are suppressed, may still show warnings elsewhere), e.g. https://goreportcard.com/report/github.com/Microsoft/hcsshim

Reply:
That said; `errcheck` specifically can be a bit of a pain for some errors that are generally safe to ignore. Still, it sometimes catches errors that should _not_ have been ignored, so explicit > implicit is good.

(sorry for the "peanut gallery" comments 😂)

Reply:
Oh, `_, _ = he.p.Process.Kill(ctx)` looks better.

Reply:
That seems a bit better to me (albeit not the greatest looking as you've said). Curious how others on the team feel though

Reply:
I definitely think we should keep this linter enabled. There's a lot of `errcheck` suppressions in this PR, but I want this linter to help encourage us to not ignore errors in the future. The least we can do is log them :)

Reply:
I think `_, _ = he.p.Process.Kill(ctx)` is fine if others like it more

Reply:
I think it looks much better, and it's nice to use standard Go syntax rather than linter comments. :)

Reply:
Updated.